server/src/main/java/org/elasticsearch/action/termvectors/TermVectorsRequest.java

```diff
@@ -519,7 +492,10 @@ public void readFrom(StreamInput in) throws IOException {
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         super.writeTo(out);
-        out.writeString(type);
+        if (out.getVersion().before(Version.V_8_0_0)) {
+            // types not supported so send an empty array to previous versions
+            out.writeString(null);
```

Comment:
hmm, this would fail wouldn't it? I thought only writeOptionalString would accept nulls? do you need to write `_doc` instead? Let's also make sure at least one test goes through this path?